dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -35,7 +38,9 @@ internal open class RumViewScope(
     internal val name: String,
     eventTime: Time,
     initialAttributes: Map<String, Any?>,
-    internal val firstPartyHostDetector: FirstPartyHostDetector
+    internal val firstPartyHostDetector: FirstPartyHostDetector,
+    internal val cpuVitalMonitor: VitalMonitor,
```

Comment:
To avoid that overflow issues we could use `BigDecimal` instead of `Double` when we do the conversion in the `CpuVitalReader`

Reply:
There's no overflow issue. Unlike the iOS issue mentioned yesterday, the cpu info starts when the application got started, and it won't reach the max value unless the app is left open for more than 24 hours which is not common in Android.